packages/qwik-city/runtime/src/server-functions.ts

```diff
@@ -380,6 +380,18 @@ export const serverQrl = <T extends ServerFunction>(
             throw obj;
           }
           return obj;
+        } else if (contentType === 'application/json') {
```

Comment:
theses cases should be hit but in cases that I mentioned it's better that they're handled